### PR TITLE
Rebase pyt_2.0_triton onto upstream master

### PR DESCRIPTION
**Pull request by jithunnair-amd**
[conflicts.txt](https://github.com/ROCmSoftwarePlatform/pytorch/files/10558445/conflicts.txt)

### CODE REVIEW

**Comment by jithunnair-amd** [comment]
Hi @dllehr-amd and @pruthvistony, I faced some conflicts which I uploaded as an attachment to this PR in the description. I also faced some tricky conflicts with `.ci/pytorch/build.sh` since `.jenkins` was moved upstream to `.ci`

File: .ci/pytorch/build.sh [hunk 1]
```diff
@@ -309,3 +309,6 @@ if [[ "$BUILD_ENVIRONMENT" != *libtorch* && "$BUILD_ENVIRONMENT" != *bazel* ]];
 fi
 
 print_sccache_stats
+
```
Comment (pruthvistony) [inline .ci/pytorch/build.sh:312]: This code wasnt reverted properly.
Can keep it or delete. it should be harmless code.

**Review by pruthvistony — APPROVED** [review]

File: .ci/pytorch/build.sh (hunk 1 above)
Comment (jithunnair-amd) [reply .ci/pytorch/build.sh]: Thanks, I deleted it to minimize diffs vs upstream.

**Comment by jithunnair-amd** [comment]
Merging this PR since @jataylo reported that the PyTorch build was successful with this branch in his local runs. The criteria for this PR is just a successful PyTorch build.

**Pull request merged**